packages/xo-web/src/common/rich-text.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import { connectStore } from 'utils'
+import { createGetObjectsOfType } from 'selectors'
+import PropTypes from 'prop-types'
+import decorate from 'apply-decorators'
+import { injectState, provideState } from 'reaclette'
+import Copiable from 'copiable'
+import { flatMap } from 'lodash'
+import Link from 'link'
+import store from 'store'
+
+const RichText = decorate([
+  connectStore({
+    vms: createGetObjectsOfType('VM'),
+    hosts: createGetObjectsOfType('host'),
+    pools: createGetObjectsOfType('pool'),
+    srs: createGetObjectsOfType('SR'),
+  }),
+  provideState({
+    computed: {
+      uuidToLink: (_, props) => {
+        const regex = /\b(?:OpaqueRef:)?[0-9a-f]{8}(?:-[0-9a-f]{4}){3}-[0-9a-f]{12}\b/g
+        const parts = props.message.split(regex)
+        const ids = props.message.match(regex) || []
+        const { objects } = store.getState()
+
+        return flatMap(parts, (part, index) => {
+          // If on last part, return only the part without adding Copiable component
+          if (index === ids.length) {
+            return part
+          }
+
+          const id = ids[index]
+          let _object
+
+          for (const collection of [props.vms, props.hosts, props.pools, props.srs]) {
+            // TODO : check permissions for objects retrieved by refs
```

Comment:
Only admins have access to `settings`

Reply:
It's a reminder in case the component is used in non-admin pages.
However, based on your recommandation, I moved the comment in a JSDoc comment on the component for better visibility.